pkgs/development/python-modules/openaiauth/default.nix

```
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, setuptools, requests, lib }:
+
+buildPythonPackage rec {
+  pname = "OpenAIAuth";
+  version = "0.3.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "CPcBgGvxRO677EdPI3lNtJXkCW7el6N6N2GeaDo5ApU=";
+  };
+
+  buildInputs = [ setuptools ];
+
+  propagatedBuildInputs = [ requests ];
+
+  meta = with lib; {
+    description = "A Python library for authenticating with the OpenAI API.";
+    license = licenses.mit;
+    maintainers = with maintainers; [ realsnick ];
+    homepage = "https://github.com/openai/openai-python/tree/master/openai/auth";
```

```suggestion
    homepage = "https://github.com/acheong08/OpenAIAuth";
```

The original link 404s for me, pypi says this link was correct.